SunEngine.Admin/Managers/CacheSettingsManager.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using LinqToDB;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Options;
+using SunEngine.Commons.Cache.CachePolicy;
+using SunEngine.Commons.Configuration.Options;
+using SunEngine.Commons.DataBase;
+using SunEngine.Commons.Models;
+using SunEngine.Commons.Services;
+using SunEngine.Commons.Utils;
+using SunEngine.Commons.Utils.CustomExceptions;
+
+namespace SunEngine.Admin.Managers
+{
+    public class CacheSettingsManager : DbService
+    {
+        private IServiceCollection serviceCollection;
+        private IServiceProvider serviceProvider;
+        
+        public CacheSettingsManager(IServiceCollection serviceCollection, IServiceProvider serviceProvider, DataBaseConnection db) : base(db)
+        {
+            this.serviceCollection = serviceCollection;
+            this.serviceProvider = serviceProvider;
+        }
+
+        public async Task UpdateCachePolicy(CacheSettings cacheSettings)
+        {
+            if(!Enum.IsDefined(typeof(CachePolicy), cacheSettings.CachePolicy))
+                throw new ArgumentOutOfRangeException();
```

Comment:
I would strongly suggest putting the message inside this exception. If in future we will log it or display somewhere - it could be unclear why such exception was thrown. It's always faster and easier look for the message in the source code.